ephyviewer/datasource/epochs.py

```diff
@@ -222,22 +290,75 @@ def fill_blank(self, method='from_left'):
                 ep_durations[ind+1] += gap/2.
                 
         
-        self._clean_and_set(ep_times, ep_durations, ep_labels)
+        self._clean_and_set(ep_times, ep_durations, ep_labels, ep_ids)
+    
+    def load(self):
+        """
+        Returns a dictionary containing the data for an epoch.
+        
+        Derived subclasses of WritableEpochSource override this method to
+        implement loading a file or importing data from objects in memory. The
+        superclass implementation WritableEpochSource.load() creates an empty
+        dictionary with the correct keys and types. It can be called from the
+        subclass implementation using super().load() if, for example, the file
+        to be loaded does not exist.
+        
+        The method returns a dictionary containing the loaded data in this form:
+        
+        { 'time': np.array, 'duration': np.array, 'label': np.array, 'name': string }
+        """
+        
+        s = max([len(l) for l in self.possible_labels])
+        epoch = {'time':     np.array([], dtype='float64'),
+                 'duration': np.array([], dtype='float64'),
+                 'label':    np.array([], dtype='U'+str(s)),
+                 'name':     self.channel_name}
+        return epoch
     
     def save(self):
         print('WritableEpochSource.save')
+        raise NotImplementedError()
     
 
 
+class CsvEpochSource(WritableEpochSource):
```

Comment:
OK for me.

I will add my ExcelEpochSource here after this PR because it is basically the same except: 
**df = pd.read_csv(self.filename,  index_col=None)** >>> df = pd.read_excel(self.filename,  index_col=None).

Reply:
Sounds great!